File: two_pointers/977_squares_of_sorted_array.go
import "math"

func sortedSquares(nums []int) []int {
	res := make([]int, len(nums))

	l := 0
	r := len(nums) - 1
	for i := len(nums) - 1; i >= 0; i-- {
		leftNum := nums[l]
		rightNum := nums[r]
		if math.Abs(float64(leftNum)) > math.Abs(float64(rightNum)) {
			res[i] = leftNum * leftNum
			l++
		} else {
			res[i] = rightNum * rightNum
			r--
		}
	}

	return res
}

File: two_pointers/977_squares_of_sorted_array.py
def sorted_squares(self, nums: list[int]) -> list[int]:
        n = len(nums)
        res = [0] * n
        
        l = 0
        r = n - 1

        for i in range(n -1, -1, -1):
            left_num = nums[l]
            right_num = nums[r]
            if abs(left_num) > abs(right_num):
                res[i] = left_num*left_num
                l += 1
            else:
                 res[i] = right_num*right_num
                 r -= 1
        return res
